Userland/Libraries/LibDiff/Generator.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021, Mustafa Quraish <mustafa@cs.toronto.edu>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Generator.h"
+
+namespace Diff {
+
+Vector<Hunk> from_text(StringView const& old_text, StringView const& new_text)
+{
+    auto old_lines = old_text.lines();
+    auto new_lines = new_text.lines();
+
+    enum Direction {
+        Up,       // Added a new line
+        Left,     // Removed a line
+        Diagonal, // Line remained the same a line
+    };
+
+    // A single cell in the DP-matrix. Cell (i, j) represents the longest common
```

Comment:
Add some comment referencing what algorithm this implementing? `DP-matrix` etc all seem like terms from the literature of some algorithm?

Reply:
Looks like you referenced it in the commit message but not the code?

Reply:
Good point, added a comment describing this.